core/commands/swarm.go

```diff
@@ -201,12 +232,31 @@ var swarmPeeringRmCmd = &cmds.Command{
 				return err
 			}
 		}
+
+		save, _ := req.Options[swarmSaveOptionName].(bool)
+		if save {
+			r, err := fsrepo.Open(env.(*commands.Context).ConfigRoot)
+			if err != nil {
+				return err
+			}
+			defer r.Close()
+			cfg, err := r.Config()
+			if err != nil {
+				return err
+			}
```

Comment:
I think we can avoid repeating all this code and separate it into a function.

Reply:
Factored out into new method `updateAndPersistConfig` in [this commit](https://github.com/ipfs/kubo/pull/9192/commits/94fc792df4ab6b858b4b9e980aa6f7ffca24bd03).